test/C/src/federated/ClockSync.lf

```diff
@@ -0,0 +1,81 @@
+/**
+ * This program tests clock synchronization. It sets an initial clock sync error of 2 s and checks
+ * that the clock synchronization error at startup and at shutdown is no larger than 10x this. This
+ * is, of course, extremely pessimistic, but the test is mainly ensuring the clock sync compiles and
+ * runs correctly. The pessimistic threshold is designed to ensure the test does not fail even with
```

Comment:
I think it is worth keeping this test because future changes to the clock sync algorithm could cause this test to fail, particularly if clock synchronization becomes unstable (it is, after all, a feedback control system).  I think that the test flaking is really extremely unlikely, so I suggest that we keep it here for now, and the first time it flakes (if ever), then we follow your suggestion and move it to examples.

Reply:
Sounds great!